core/src/main/java/org/eclipse/hono/util/QoS.java

```diff
@@ -0,0 +1,51 @@
+/*******************************************************************************
+ * Copyright (c) 2016, 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.util;
+
+/**
+ * Denotes the QoS level with which a message was sent by the device.
+ */
+public enum QoS {
+
+    AT_MOST_ONCE,
+    AT_LEAST_ONCE,
+    /**
+     * Indicates the QoS is not settable by the device.
+     */
+    NOT_APPLICABLE,
+    /**
+     * Indicates that the device set a QoS level which is not known or supported.
+     */
+    UNKNOWN;
+
+    /**
+     * Derive the QoS level from the given number.
```

Comment:
Makes sense, yes this is used for MQTT only.

Reply:
After a closer look it turned out that this method was in fact used in the HTTP adapter. The HTTP adapter accidentally (I guess) shares its QoS level definition with MQTT.

I think it makes more sense to have this in the corresponding Context implementations, since it is protocol specific.